Linux knowledge graph initial version.

jdijt opened the pull request:
This is still rough and will get updated, but creating pull request to make it more visible for people to look at. Feedback is welcome :)

dirkjonker commented on core-curriculum/knowledge/linux.json:
```diff
@@ -0,0 +1,225 @@
+{
+    "id": "linux",
+    "name": "Linux",
+    "description": "Linux is an operating system dat is commonly used on server systems. This subject is focused on specifically that Linux use case.",
```
take a closer look at the wording of this sentence

dirkjonker commented on core-curriculum/knowledge/linux.json:
```diff
@@ -0,0 +1,225 @@
+            "abilities":[
+                "Navigating the directory structure of a Linux box from the terminal.",
+                "Search for a specific file by name within the file system.", 
+                "Search trough file(s) by content.",
```
through

dirkjonker commented on core-curriculum/knowledge/linux.json:
```diff
@@ -0,0 +1,225 @@
+        {
+            "id": "processmanagement",
+            "name": "Managing Processes",
+            "description":"On a linux machine processes run in a tree, always having an \"init\" process at the root that is owned by the root user. There exist various tools to inspect and manage this tree of processes. In this topic we discuss the process tree, various tools to explore it, how to manage processes, how process ownership works and ",
```
this sentence is not finished

dirkjonker commented on core-curriculum/knowledge/linux.json:
```diff
@@ -0,0 +1,225 @@
+            "dependencies":["basic"],
+            "abilities": [
+                "List the various types of (pseudo)files that can exist in the Linux system.",
+                "Explain the purpose of the directories present at the root level of most linux filesystem."
```
maybe something about mounts?
for people coming from windows?

dirkjonker commented on core-curriculum/knowledge/linux.json:
```diff
@@ -0,0 +1,225 @@
+                "Generating ssh private keys.",
+                "Setting up a port forward on an SSH tunnel.",
+                "Execute commands directly on remote host.",
+                "Understand the limitations of SSH (e.g.: what happens if I start a long running migration in an SSH session and my connection breaks)"
```
maybe add the usage of tmux (or screen)?

dirkjonker commented on core-curriculum/knowledge/linux.json:
```diff
@@ -0,0 +1,225 @@
+            "description":"There are almost as many different package managers as there are linux distributions. However they share many common operations, just with a different syntax. The goal of this topic is to get an understanding of the common operations: installing, removing, upgrading.",
+            "tags":["required-for-junior"],
+            "dependencies":["basic"],
+            "abilities": [],
```
since usually you will encounter either Debian or Red Hat can we limit it to `apt` and `yum`/`dnf`?

jdijt replied:
Seems like a good idea, maybe also add a link to the "package management rosetta" by Arch Linux (https://wiki.archlinux.org/index.php/Pacman/Rosetta ).
I was thinking of also including  Alpine btw, as many lightweight docker images are based on that.

dirkjonker commented on core-curriculum/knowledge/linux.json:
```diff
@@ -0,0 +1,225 @@
+        {
+            "id": "servicemanagement",
+            "name": "Managing Services",
+            "description":"Common tools for managing services on Linux, primarily systemd.",
```
can we turn systemd into its own topic? As it is used by pretty much every distribution right now

jdijt replied:
I'd prefer that people learn about the principle behind service management, i.e.: some process that is responsible for making sure all services start and the right order that has to be configged and then use SystemD as a leading example.

Then we can add a follow-up topic like "advanced service management with systemd".

eamelink commented:
This is extremely well done. The graph looks good! 👍 

The two things that I noticed while going through it are:
* Networking 1 - I doubt if the third ability; using iptables to open and close ports should be required-for-junior
* Maybe the first three abilities from 'Using Bash' should be required-for-junior

Also, how about we sit together for a while to come up with a bunch of assessment questions?

jdijt commented:
So I took IPTables out of networking 1.

I also split up disk management further into a "inspecting" node, for investigating/diagnosing problems and two disk management nodes, 1 for a basic setup (formatting/partitioning), one for going wild with devicemapper / RAID.

Similar for service management, split it up in the basic administering task and a more advanced 'defining your own services' node.

Added a 'performance debugging' node, and added a 'networked filesystems' node.

Finally added at least 1 resource to each topic.

nMoncho commented:
Nice work, very thorough.
Just being nitpick here, can you make the ids in the same format as all others, separating words with hyphens?
